### ISSUE TITLE
Validation loss increases

### ISSUE THREAD

**Issue by Spiruel** [sentinel-hub/field_delineation#17]
Hi,

I'm struggling to train a field delineation model successfully.

I am using field parcels in Belgium, and have processed the npz files properly. The input files seem to be correct, with aligned distance, boundary, extent labels, appropriate Sentinel 2 imagery etc.
eg.
![download](https://user-images.githubusercontent.com/3259632/194285649-5827dc55-86bd-47f2-81c4-eb6a272b4ce5.png)

However when I fit the model (3 folds), my training losses all decrease while my validation losses all increase. My wandb run is hosted here: https://wandb.ai/spiruel/field-delineation/runs/2pa2y8rs?workspace=user-spiruel

Please can you offer some suggestions why this might be happening, and what I can try to fix this?

Here is an example of my model predictions on the failed run:

![download (1)](https://user-images.githubusercontent.com/3259632/194286530-9e11b269-1dd8-479b-a956-9e2f6eac9c4b.png)

Thanks

```
eo-flow @ git+https://github.com/sentinel-hub/eo-flow@d7203fb20bb402857765327244a9448b6105568d
eo-learn==1.2.1
```

**Comment by Spiruel** [comment]
I've since discovered a previous issue https://github.com/sentinel-hub/field-delineation/issues/14 and I'm working through the suggestions there.

I've attached some example npz files 
[patchlets_field_delineation_.zip](https://github.com/sentinel-hub/field-delineation/files/9726465/patchlets_field_delineation_.zip)

I can confirm the two channels of the predicted arrays (extent/boundary) and sum to ~1

I have shuffled the npz files so there's no eopatch/location-dependent ordering coming into play.

**Comment by devisperessutti** [comment]
Hi @Spiruel 

Did you manage to make progress on this?

As it relates more to the optimisation of ML pipeline rather than the code itself I would close the issue.

**Comment by Spiruel** [comment]
Hi @devisperessutti 

Thanks for your reply. Unfortunately I've still not cracked it. It would still be really useful to me if you're able to double check my example npz files and confirm whether it is a data issue on my end, or a wider code issue. 

I'll be having another go at this week and hope to update then. If not me updating on this thread, then perhaps it would be possible to diagnose the issue over email with a few quick pointers from you, time permitting?

 Thanks

**Comment by devisperessutti** [comment]
Hi @Spiruel 

The patchlets you sent look as I would expect. Did you modify perhaps some parameters in the rasterization or in some other workflow?

I would give the same suggestions as in #14, and try to apply the changes one at time. How many patchlets do you have overall?

It seems like the model is predicting something, although the extent probabilities look more like boundaries should, and viceversa. Is there maybe a swapping of masks occurring somewhere?

**Comment by batic** [comment]
Closing issue due to inactivity.
